docs/specification.md

```diff
@@ -1,49 +1,190 @@
+All gittuf-specific metadata and information are tracked in a separate Git ref,
+`refs/gittuf`.
+
+### Reference State Log (RSL)
+
+Note: This document presents only a summary of the academic paper and a
+description of gittuf's implementation of RSL. A full read of the paper is
+recommended.
+
+The Reference State Log contains a series of entries that each describe some
+change to a Git ref. Each entry contains the ref being updated, the new location
+it points to, and a hash of the parent RSL entry. The entry is signed by the
+actor making the change to the ref.
+
+Given that each entry in effect points to its parent entry using its hash, an
+RSL is a Merkle tree. gittuf's implementation of the RSL uses Git's underlying
+Merkle graph. Generally, gittuf is designed to ensure the RSL is linear but a
+privileged attacker may be able to cause the RSL to branch, resulting in a
+forking attack.
+
+The RSL is tracked at `refs/gittuf/reference-state-log`, and is implemented as a
+distinct commit tree. Each commit in this tree corresponds to one entry in the
+RSL. The commit message has a fixed format `<ref name>: <commit ID>`, and the
+commit is signed using the actor's key.
+
+### Actor Access Control Policies
+
+Note: This section assumes some prior knowledge of the TUF specification.
+
+There are several aspects to how defining the access privileges an actor has.
+First, actors must be established in the repository unambiguously, and gittuf
+uses TUF's mechanisms to associate actors with their signing keys. TUF metadata
+distributes the public keys of all the actors in the repository and if a key is
+compromised, new metadata is issued to revoke its trust.
+
+Second, TUF allows for defining _namespaces_ for the repository. TUF's notion of
+namespaces aligns with Git's, and TUF namespaces can be used to reason about
+both Git refs and files tracked within the repository. Namespaces are combined
+with TUF's _delegations_ to define sets of actors who are authorized to make
+changes to some namespace. As such, the owner of the repository can use gittuf
+to define actors representing other contributors to the repository, and delegate
+to them only the necessary authority to make changes to different namespaces of
+the repository.
+
+Policies for gittuf access are defined using a subset of TUF roles. The owners
+of the repository hold the keys used to sign the Root role that delegates trust
+to the other roles. The top level Targets role and any Targets roles it
+delegates to contain restrictions on protected namespaces. The specifics of the
+delegation structure vary from repository to repository as each will have its
+own constraints.
+
+A typical TUF delegation connects two TUF Targets role. In the delegations
+graph where each node is a Targets role, delegations are the edges connecting
+them. The delegations graph is traversed when verifying some target until the
+leaf node is encountered, after which the target's entry is compared against the
```

Comment:
This is a little confusing.  There really isn't a meaningful notion of leaf nodes as you may have cycles in delegations and may also have intermediate nodes that directly list targets in addition to delegating.

Reply:
I'll take another pass. gittuf has a very opinionated approach to TUF delegations. Users are not expected to hand-craft metadata and we always expect the entries to be the commits themselves or the RSL entry so we shouldn't have the intermediate metadata recording hashes issue.

Reply:
What do you mean that entries are the commits themselves?  I thought they were either a path, branch, or tag?

Reply:
I've updated this part of the document to try and make it clearer.

Reply:
I would tweak the new text to more state that TUF's namespacing is general and can be used for various git namespaces like tags, files, branches, etc.

Reply:
Mentioned namespaces generally and said gittuf applies it to git but modifies the overall workflow to incorporate RSLs etc.